utils/config.mlp

```diff
@@ -65,9 +65,9 @@ let mkdll, mkexe, mkmaindll =
           let c = flexlink.[i] in
           if c = '/' then '\\' else c in
         (String.init (String.length flexlink) f) ^ " %%FLEXLINK_FLAGS%%" in
-      flexlink,
+      flexlink ^ "%%FLEXLINK_DLL_LDFLAGS%%",
       flexlink ^ " -exe%%FLEXLINK_LDFLAGS%%",
-      flexlink ^ " -maindll"
+      flexlink ^ " -maindll%%FLEXLINK_DLL_LDFLAGS%%"
```

Comment:
Don't you need a space after `-maindll`?  Maybe not because you put a space at the start of  `%%FLEXLINK_DLL_LDFLAGS%%` when it's not empty, but this is confusing, and a space here would be so much clearer.

Reply:
The space does indeed come from `%%FLEXLINK_DLL_LDFLAGS%%`. The whole `flexlink` thing is a bit of a mess anyway - we keep hitting elsewhere confusion/problems because `mkdll`, `mkexe` and `mkmaindll` are usually C compilers, but `flexlink` is a linker. In the relatively short-term, I'm hoping to change the way the driver works so that `flexlink` ceases to be a wrapper around the linker (and instead is responsible for generating object files to be linked with the program in the normal way), at which point all this configuration evil disappears completely.

Would it be OK therefore to leave this one as-is for now (vaguely for "consistency" with the same evil already present for `-exe`)?

Reply:
I don't see why there is no space in `" -exe%%FLEXLINK_LDFLAGS%%"` either :-)  But I'll close my eyes.